imap_data_access/file_validation.py

```diff
@@ -444,11 +539,142 @@ def construct_path(self) -> Path:
             Upload path
         """
         spice_dir = imap_data_access.config["DATA_DIR"] / "spice"
-        subdir = _SPICE_DIR_MAPPING[self.file_extension]
+        subdir = _SPICE_DIR_MAPPING[self.spice_metadata["type"]]
         # Use the file suffix to determine the directory structure
         # IMAP_DATA_DIR/spice/<subdir>/filename
         return spice_dir / subdir / self.filename
 
+    @staticmethod
+    def _handle_doy(part, components):
```

Comment:
This was confusing to me at first because we are modifying objects in-place that were passed in (including deleting them), catching errors and returning strings. There are a lot of moving pieces within this function that I think could get inconsistent very easily if errors happen partway through.

I think this all could be much simpler if we do the parsing of the date on the value in-line below. Let the datetime ValueErrors propagate up if they happen.

Reply:
I initially did that in-line, but the pre-commit checks complained I had too many if statements lol. I'll see if I can reduce the number

Reply:
Don't feel like we have to obey our precommit overlords. If it makes sense to you to do that, then feel free to add a noqa block and say why you chose that.